vendor/github.com/docker/libnetwork/drivers/overlay/joinleave.go

```diff
@@ -145,6 +145,21 @@ func (d *driver) Join(nid, eid string, sboxKey string, jinfo driverapi.JoinInfo,
 	return nil
 }
 
+func (d *driver) DecodeTableEntry(tablename string, key string, value []byte) (string, string) {
+	if tablename != ovPeerTable {
+		logrus.Errorf("DecodeTableEntry: unexpected table name %s", tablename)
+		return "", ""
+	}
+
+	var peer PeerRecord
+	if err := proto.Unmarshal(value, &peer); err != nil {
+		logrus.Errorf("DecodeTableEntry: failed to unmarshal peer record for key %s: %v", key, err)
+		return "", ""
+	}
+
+	return key, "Host IP: " + peer.TunnelEndpointIP
```

Comment:
We want the driver to return the endpointID and a string that can be presented to the user. The reason for this is the way networkDB API is currently designed. NetworkDB is a service provided by the libnetwork and the network drivers use it. But libnetwork is not aware of what exactly is the key used by the driver. Please see this https://github.com/docker/libnetwork/pull/1674#discussion_r104831757.